client/components/date-range/use-shortcuts.ts

```diff
@@ -1,133 +1,138 @@
 import config from '@automattic/calypso-config';
 import { translate, useTranslate } from 'i18n-calypso';
+import { memoize } from 'lodash';
 import { getMomentSiteZone } from 'calypso/my-sites/stats/hooks/use-moment-site-zone';
 import { useSelector } from 'calypso/state';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
-import { DateControlPickerShortcut } from '../date-control/types';
+import { DateRangePickerShortcut } from './shortcuts';
 
 export const DATERANGE_PERIOD = {
+	HOUR: 'hour',
 	DAY: 'day',
 	WEEK: 'week',
 	MONTH: 'month',
 };
 
+const DATE_FORMAT = 'YYYY-MM-DD';
+
 export function getShortcuts(
 	state: object,
 	dateRange: {
 		chartStart: string;
 		chartEnd: string;
 		daysInRange: number;
 	},
-	shortcutList?: string[],
 	translateFunction = translate,
 	isNewDateFilteringEnabled = config.isEnabled( 'stats/new-date-filtering' )
 ) {
 	const siteId = getSelectedSiteId( state );
 	const siteToday = getMomentSiteZone( state, siteId );
+	const siteYesterday = siteToday.clone().subtract( 1, 'days' );
+	const siteTodayStr = siteToday.format( DATE_FORMAT );
+	const yesterdayStr = siteYesterday.format( DATE_FORMAT );
 
-	const supportedShortcutList: DateControlPickerShortcut[] = [
-		{
-			id: 'last_7_days',
-			label: translateFunction( 'Last 7 Days' ),
-			offset: 0,
-			range: 6,
-			period: DATERANGE_PERIOD.DAY,
-			shortcutId: 'last_7_days',
-		},
-		{
-			id: 'last_30_days',
-			label: translateFunction( 'Last 30 Days' ),
-			offset: 0,
-			range: 29,
-			period: DATERANGE_PERIOD.DAY,
-			shortcutId: 'last_30_days',
-		},
-		{
-			id: 'last_3_months',
-			label: translateFunction( 'Last 90 Days' ),
-			offset: 0,
-			range: 89,
-			period: DATERANGE_PERIOD.WEEK,
-			shortcutId: 'last_3_months',
-		},
-		{
-			id: 'last_year',
-			label: translateFunction( 'Last Year' ),
-			offset: 0,
-			range: 364, // ranges are zero based!
-			period: DATERANGE_PERIOD.MONTH,
-			shortcutId: 'last_year',
-		},
-		{
-			id: 'custom_date_range',
-			label: translateFunction( 'Custom Range' ),
-			offset: 0,
-			range: 0,
-			period: DATERANGE_PERIOD.DAY,
-			shortcutId: 'custom_date_range',
-		},
-	];
+	return memoize(
```

Comment:
`memoize` seems to be used wrong 🤔

Reply:
yeah it was wrong. it acts like a decorator and the step should be done outside of the selector.